File: app/src/main/java/com/plbear/iweight/base/BasePreferenceActivity.java
package com.plbear.iweight.base;

import android.os.Bundle;
import android.preference.PreferenceActivity;
import android.support.annotation.Nullable;

import com.plbear.iweight.utils.LogInfo;

import butterknife.ButterKnife;

/**
 * Created by yanyongjun on 2018/6/11.
 */

public class BasePreferenceActivity extends PreferenceActivity {
    protected String TAG = getClass().getSimpleName();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AppManager.getAppManager().addToStack(this);
        ButterKnife.bind(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        AppManager.getAppManager().removeFromStack(this);
    }

    protected void loginfo(String info) {
        LogInfo.i(TAG, info);
    }

    protected void logerror(String error) {
        LogInfo.e(TAG, error);
    }

}

File: app/src/main/java/com/plbear/iweight/base/BasePreferenceActivity.kt
package com.plbear.iweight.base

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.preference.PreferenceActivity
import com.plbear.iweight.utils.LogInfo

/**
 * Created by yanyongjun on 2017/normal_4/normal_1.
 */

class BasePreferenceActivity : PreferenceActivity() {
    protected val TAG = this.javaClass.simpleName

    internal var mReceiver: BroadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context?, intent: Intent?) {
            if (intent == null || context == null) {
                logerror("onReceive context == null or intent == null")
            }
            val action = intent!!.action
            loginfo("onReceive action:" + action!!)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        loginfo("onCreate")
        AppManager.getAppManager().addToStack(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        AppManager.getAppManager().removeFromStack(this)
    }

    fun loginfo(info: String) {
        LogInfo.i(TAG, info)
    }

    fun logerror(error: String) {
        LogInfo.e(TAG, error)
    }
}
